lms/djangoapps/courseware/access.py

```diff
@@ -136,8 +136,9 @@ def has_access(user, action, obj, course_key=None):
     if not user:
         user = AnonymousUser()
 
-    if in_preview_mode():
-        if not bool(has_staff_access_to_preview_mode(user=user, obj=obj, course_key=course_key)):
+    # Preview mode is only accessible by staff.
+    if in_preview_mode() and course_key:
```

Comment:
If in_preview_mode() should never return True if course_key is None, then you don't need to check that course_key is not None on line 140. The unnecessary check could end up masking a coding bug. I would consider this a Nit though.